collector/cpu_netbsd.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !nocpu
+// +build !nocpu
+
+package collector
+
+import (
+	"errors"
+	"math"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"unsafe"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sys/unix"
+
+	"howett.net/plist"
+)
+
+type clockinfo struct {
+	hz     int32 // clock frequency
+	tick   int32 // micro-seconds per hz tick
+	spare  int32
+	stathz int32 // statistics clock frequency
+	profhz int32 // profiling clock frequency
+}
+
+type cputime struct {
+	user float64
+	nice float64
+	sys  float64
+	intr float64
+	idle float64
+}
+
+type plistref struct {
+	pref_plist unsafe.Pointer
+	pref_len   uint64
+}
+
+type values struct {
```

Comment:
values and prop(s) are a bit too generic for package wide variables that are only used here. Can you give it a bit more specific name? Or do you think they are useful beyond the cpu collector?

Reply:
I agree, these names are a bit generic to be package level with no context.

Also, it would be good to document these structs.

```go
// values is something something something.
type values struct {
```